### PR TITLE
[Improve](point query) Improve lookup connection cache from DoubleBuffer to LRU cache for better item pruning

### PR DESCRIPTION
**Pull request by eldenmoon**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.14 seconds
 stream load tsv:          504 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17168959829 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.07 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17164304144 Bytes

File: be/src/service/point_query_executor.h [hunk 1]
```diff
@@ -186,76 +189,59 @@ class RowCache {
...
+    void add(__int128_t cache_id, std::shared_ptr<Reusable> item) {
+        std::string key = encode_key(cache_id);
+        CacheValue* value = new CacheValue;
+        value->last_visit_time = UnixMillis();
+        value->item = item;
+        auto deleter = [](const doris::CacheKey& key, void* value) {
+            CacheValue* cache_value = (CacheValue*)value;
+            delete cache_value;
+        };
+        LOG(INFO) << "Add item mem size " << item->mem_size()
```
Comment (xinyiZzz) [inline be/src/service/point_query_executor.h:219]: Could this log be too frequent
Maybe add metrics to `add` and `get`

**Review by xinyiZzz — APPROVED** [review]
LGTM

File: be/src/service/point_query_executor.h (hunk 1 above)
Comment (eldenmoon) [reply be/src/service/point_query_executor.h:219]: item add is very rare , so not too much log will be printed

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**